src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -626,17 +631,23 @@ private struct UrlDecoder
             private void FlushBytes()
             {
                 Debug.Assert(_numBytes > 0);
+                if (_charBuffer == null)
+                    _charBuffer = new char[_bufferSize];
+
                 _numChars += _encoding.GetChars(_byteBuffer, 0, _numBytes, _charBuffer, _numChars);
                 _numBytes = 0;
             }
 
             internal UrlDecoder(int bufferSize, Encoding encoding)
             {
+                _containsUnsafe = false;
+                _containsSpaces = false;
+
                 _bufferSize = bufferSize;
                 _encoding = encoding;
 
-                _charBuffer = new char[bufferSize];
-                
+                _charBuffer = null; // char buffer created on demand
```

Comment:
I see, ok.  (Seems strange then that we're lazily allocating the byte[] array; doing so suggests we don't expect any ASCII chars in common input, but let's not change it.)